ibis/backends/duckdb/__init__.py

```diff
@@ -1156,7 +1200,30 @@ def fetch_from_cursor(
                 for name, col in zip(table.column_names, table.columns)
             }
         )
-        return PandasData.convert_table(df, schema)
+        df = PandasData.convert_table(df, schema)
+        if not df.empty and geospatial_supported:
+            return self._to_geodataframe(df, schema)
+        return df
+
+    # TODO(gforsyth): this may not need to be specialized in the future
+    @staticmethod
+    def _to_geodataframe(df, schema):
+        """Convert `df` to a `GeoDataFrame`.
+
+        Required libraries for geospatial support must be installed and
+        a geospatial column is present in the dataframe.
+        """
+        import geopandas as gpd
+
+        geom_col = None
+        for name, dtype in schema.items():
+            if dtype.is_geospatial():
+                if not geom_col:
+                    geom_col = name
+                df[name] = gpd.GeoSeries.from_wkb(df[name])
```

Comment:
Quick explainer why we can't use the base implementation of this:

the base `to_geodataframe` uses shapely to take the output from postgis (or others), and converts it to a format compatible with a `geopandas.GeoArray`.  We already have the outputs from DuckDB spatial as a WKB, so we 
a) don't need to use shapely and 
b) can't use the base version because shapely gets angry if you try to `to_shape` on a WKB

Reply:
Just double checking, in case  I misunderstood.  Do we want to leave a comment in the code about this, or the comment in GH is just fine. ?

Reply:
the GH comment is good, I don't think we need something inline